src/Dao.jsx

```diff
@@ -757,11 +770,143 @@ const Dao = () => {
       return string.match(urlRegex);
     }
   }
-
   const submitProposal = async (e) => {
     e.preventDefault();
     e.persist();
 
+    
+    // Handle roketo stream creation
+    if (e.target.name === 'newProposalRoketoStream') {
+      // const argsList = {
+      //   args: {
+      //     owner_id: e.target.proposalTokenOwner.value.trim(),
+      //     total_supply: new Decimal(e.target.proposalTokenSupply.value.trim()).mul(new Decimal(10).pow(e.target.proposalTokenDecimals.value)).round(0, 0).toFixed(0),
+      //     metadata: {
+      //       spec: "ft-1.0.0",
+      //       name: e.target.proposalTokenName.value.trim(),
+      //       symbol: e.target.proposalTokenSymbol.value.trim(),
+      //       icon: e.target.proposalTokenIcon.value.trim(),
+      //       decimals: '^' + e.target.proposalTokenDecimals.value + '^',
+      //     },
+      //   },
+      // }
+      const nearAccountValid = await accountExists(proposalTarget.value);
+      let validateTarget = validateField("proposalTarget", proposalTarget.value);
+      let validateDescription = validateField("proposalDescription", proposalDescription.value);
+      let validateSpeed = validateField("proposalRoketoSpeed", proposalRoketoSpeed.value);
+
+
+      let validatePaymentOption = true
+      if (paymentOption === "FT") {
+        validatePaymentOption = validateField("proposalFT", e.target.proposalFT.value);
+      }
+
+      let ftMetadata = null;
+      if (paymentOption === "FT") {
+        try {
+          const tokenContract = new Contract(window.walletConnection.account(), e.target.proposalFT.value, {
+            viewMethods: ['ft_metadata'],
+            changeMethods: [],
+          })
+          ftMetadata = await tokenContract.ft_metadata({})
+        } catch (error) {
+          // nu vse
+          console.log(error);
+        }
+      }
+
+      console.log("FTR", {
+        ftMetadata,
+        validateSpeed,
+        willpass:validateTarget && nearAccountValid && validateDescription && validateSpeed && validatePaymentOption && (paymentOption === "FT" && ftMetadata) || (paymentOption === "NEAR" && !ftMetadata)
+      })
+      if (validateTarget && nearAccountValid && validateDescription && validateSpeed && validatePaymentOption && (paymentOption === "FT" && ftMetadata) || (paymentOption === "NEAR" && !ftMetadata)) {
+        const amount = new Decimal(e.target.proposalAmount.value);
+        const isFt = paymentOption === "FT";
+
+        // Its a roketo testnet address, better move it to environments 
+        const roketoContractAddress = 'dev-1635510732093-17387698050424';
```

Comment:
Can you use the same contracts for consistency?

Reply:
If you are using (deploying) your own sputinkdao smart contract you better to fork the repo and deploy UI to your own GitHub or other service, as this is a reference / mockup implementation which uses specific smart contract on testnet and mainnet for testing. There is no option (yet) in this tool to select or switch smart contracts (factory) I'm afraid, so if you need to use your own, its your only option for now.

In this case, you don't need to make a PR.